spec/ddtrace/contrib/http/request_spec.rb

```diff
@@ -334,4 +334,82 @@ def expect_request_without_distributed_headers
       end
     end
   end
+
+  describe 'request exceptions' do
+    subject(:response) do
+      begin
+        client.get(path)
+      # rubocop:disable Lint/UselessAssignment
+      rescue => e
```

Comment:
Is `rescue` the right thing to do here? If we're doing this so that the exception raised by `response` doesn't crash the test, then we can instead do `expect { response }.to raise_error` or `expect { response }.to raise_error(custom_error)` instead of `expect(response).to be nil`.

Reply:
that makes much more sense, i've updated these to test for error's being raised.

I also noticed the I had a misleading spec in there testing `after_request` behavior , but in fact `after_request` wouldn't get invoked in this use case, as the error would be raised before we're able to invoke `after_request`, and adding it to get called in an `on_error` handler wouldn't be clean right now since the arguments it's expecting wouldn't be available in the on_error handler's context. Anyways I removed the spec for simplicities sake.